python/tests/test_1.cnf

```diff
@@ -0,0 +1,999 @@
+c ind 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 0
```

Comment:
I actually would prefer not having a hard-coded test case like this. I prefer generating test-cases using Python. They tend to have two advantages: (1) we remember later what they were about, how they were generated, etc. (2) they can be randomized and so they can catch multiple bugs. Would it be possible to remove this CNF from the PR? I can promise to help write a python test case generator :)

Reply:
Absolutely! I can swap the test case in the ApproxMC bindings as well. This one was mostly to just verify a "real" cnf against the bindings and make sure that nothing broke. I'm actually quite unsure how to write test cases for this type of bindings, as I mostly just turned on the verbose function in the tests and verified that the output was as expected. In fact, for the Unigen tests I don't even verify that the samples are the exact same that the command line tool returns, as I don't want them to break when changes are made to the implementation. (I did do this for ApproxMC, which I should probably change)

Since we're really testing that the interface works as expected, i.e. that the Python bindings get the same results as using the Unigen library and that nothing crashes, I don't really know how to test that without running Unigen by hand and then comparing the results to the bindings output. Any automated random tests would seem to require the bindings to be correct to verify the bindings. Please let me know if you have any ideas though, as I'd definitely like to add some more comprehensive tests!

Reply:
Hmmm.. good point. How about this minimal test case:

```
p cnf 100 1
1 2 0
```

We then ask for 1000 samples. We can then verify that the examples contain all variables 3..100 in approximately equal literal distribution of 50-50%, and 1 and 2 we should see a distribution of 66%/33% slanted towards the positive. That will be a super-simple CNF that I think is easy to reason about. What do you think?

Reply:
Great idea! I just added a similar test, but I kept the number of variables to 10 instead of 100 to keep the runtime reasonable. It currently does 1000 samples and expects everything to be within .2 of the correct distribution, though these can be tweaked at the top of the file. This is with epsilon=0.01, delta = 0.001, and kappa = 0.1. All together this runs reasonably quickly and seems indicate correct results, though I think the lower limit on sampling tolerance will keep us from getting much tighter results consistently. Let me know what you think!